common/src/main/java/net/draycia/carbon/common/messages/CarbonMessageSource.java

```diff
@@ -151,7 +148,16 @@ private void reloadTranslations() throws IOException {
     private void readLocale(final Path localeDirectory, final Path localeFile, final Locale locale) {
         this.logger.info("Found locale {} ({}) in: {}", locale.getDisplayName(), locale, localeFile);
 
-        final Properties properties = new Properties();
+        final Properties properties = new Properties() {
+            @Override
+            public synchronized Set<Map.Entry<Object, Object>> entrySet() {
+                return Collections.synchronizedSet(
```

Comment:
By creating a new collection, this breaks the ability to remove entries through entrySet, and the live-view aspect of the set, but from what I see we don't use that, so should be fine. Given that, probably want to just use unmodifiableSet instead of synchronizedSet. (If it was still a view, the double synchronization also wouldn't be needed)